### DISCUSSION TITLE
Push notifications

### DISCUSSION THREAD

**Discussion by szakes1** [appwrite/appwrite#1623] [Q&A] [answered]
Hey!

I'm curious if you're planning to implement push notifications just like in Firebase.

**Comment by eldadfux** [comment, marked as answer]
Yes we do, and we'll share an RFC for it as soon as we can.

**Reply by simpros** [reply]
Any updates on this? :)

**Reply by ConManNetwork** [reply]
I'm wondering too. Last phase of my app that can be pushed to Stores only need's mobile push notifications. Other than that Appwrite is a pure dream.

**Reply by stnguyen90** [reply]
> I'm wondering too. Last phase of my app that can be pushed to Stores only need's mobile push notifications. Other than that Appwrite is a pure dream.

@ConManNetwork I don't think we'd be able to release this in the next month 😕 so you might want to use Appwrite Functions integrated with a push notification service if you need to release push notifications soon.

**Reply by stnguyen90** [reply]
Related Discussion: 

* https://github.com/appwrite/appwrite/discussions/905

**Reply by issasaleh** [reply]
We are close to the end of 2023, and there is nothing new from you on this subject.

**Reply by cholasimmons** [reply]
it's open source you can always jump in and implement things yourself where need be.
The team already does an immense amount of work for us at such a generous fee of $0!!!

**Reply by minecrawler** [reply]
@cholasimmons even though it's opensource, the Appwrite team wants to make money. This isn't the OpenRazer tracker. They finally announced their price plan and will require me to pay for anything bigger than a PoC or toy as soon as a few weeks from now.

If I pay money, I want to have certain things in place, like professional comminication, a roadmap, expectation management. This topic has been in the works for several years with "soon we'll have something", and that's sends the wrong signals

**Comment by IzenDev** [comment]
> Yes we do, and we'll share an RFC for it as soon as we can.

Any updates?

**Reply by eldadfux** [reply]
Nothing to share yet, but the team is planing to make some progress on this during Hacktoberfest and collaborate with community members to build different adapters like we do for services like Auth and Storage.

**Reply by Spectrumtechno** [reply]
any update Push notifications?

**Comment by neiljaywarner** [comment]
@eldadfux  how did it go with this in hacktoberfest?  any thoughts on workaround.. and if realtime database sync works then local notificaitons would serve most of the needs right?

**Comment by ghost** [comment]
Hey, can we have news about push notifications please ? 🙏 It's been years now

**Comment by tk-Ravi** [comment]
Any Updates?

**Comment by stevebrainng** [comment]
This is February 2023, Any Update yet?

**Comment by navin-hariharan** [comment]
Still waiting for updates on this!

**Comment by vincent-cahart** [comment]
March 2023. Any update on this please ? 👉👈

**Comment by HeavyMedaI** [comment]
Still no updates on this?

**Reply by rafagazani** [reply]
If you want here is an example of a cloud function for appwrite:
https://github.com/open-runtimes/examples/tree/main/dart/send_push_notification

**Reply by Ryan-vdw** [reply]
> If you want here is an example of a cloud function for appwrite:
> https://github.com/open-runtimes/examples/tree/main/dart/send_push_notification

This is how I managed to get it running on my server. It's not ideal. It will still be free.

**Comment by usmandilmeer** [comment]
May 2023. Any update on Push Notifications? '🤔

**Reply by HeavyMedaI** [reply]
You may use this method apparently: https://github.com/appwrite/appwrite/discussions/1623#discussioncomment-5461238

**Comment by joeyouss** [comment]
Hello all!
Please be assured our team has this in vision and is working on it, more updates to follow once we are done working on it!

**Comment by msbjhandeer** [comment]
Any update on push notification.. i cannot use FCM as i have been implementing it in environment with no internet connection permission.. any suggestion??

**Comment by msbjhandeer** [comment]
Sure I believe that... Thanks for response.. I am new in using appwrite..
Once I finish project am doing.. I will surely look into this issue and
will love to create a hook...

**Comment by xuelink** [comment]
any updates? Sep 15, 2023

**Reply by tessamero** [reply]
There will be an update soon on this. Our engineering team is still working on this feature. Stay tuned. Updates/announcements can be found in our Discord server as well.

Also coming soon, a public roadmap to help improve transparency. 😃

**Reply by xuelink** [reply]
still waiting for this update ;)

**Comment by lantuyan** [comment]
Ahihu , i want push notification... 01/12

**Comment by lacksonmunthali** [comment]
1 December 2023, any update on this?

**Comment by p4-k4** [comment]
ETA on this?

**Comment by emilepavie** [comment]
any update on this one too?? @stnguyen90

**Comment by gewenyu99** [comment]
@emilepavie @p4-k4 @lacksonmunthali @lantuyan @behics @msbjhandeer 

Next release will include Messaging, and with it, push notifications.

You can track progress by spying on our PRs ;)

**Reply by xuelink** [reply]
I am looking forward to, still trying to handle/fix minor bugs in my appwrite fcm function. 
If its gonna be released within a week, not gonna focus here anymore haha :)
